verilog/formatting/token_annotator_test.cc

```diff
@@ -5187,19 +5187,123 @@ TEST(TokenAnnotatorTest, OriginalSpacingSensitiveTests) {
           "+",
           {NodeEnum::kDimensionScalar},
           {NodeEnum::kDimensionScalar},
-          {1, SpacingOptions::Undecided},
+          {0, SpacingOptions::Undecided},
       },
       {
           // [a  +b]
           DefaultStyle,
           verilog_tokentype::SymbolIdentifier,
           "a",
-          " ",  // 2 spaces originally
+          "  ",  // 2 spaces originally
+          '+',
+          "+",
+          {NodeEnum::kDimensionScalar},
+          {NodeEnum::kDimensionScalar},
+          {0, SpacingOptions::Undecided},  // no spacing
+      },
+      {
+          // [a     :    b]
+          DefaultStyle,
+          verilog_tokentype::SymbolIdentifier,
+          "a",
+          "     ",
+          ':',
+          ":",
+          {NodeEnum::kDimensionRange},
+          {NodeEnum::kDimensionRange},
+          {0, SpacingOptions::Undecided}, // No spacing in bit slicing/indexing
+      },
+      {
+          // [a     :    b]
+          DefaultStyle,
+          ':',
+          ":",
+          "    ",
+          verilog_tokentype::SymbolIdentifier,
+          "b",
+          {NodeEnum::kDimensionRange},
+          {NodeEnum::kDimensionRange},
+          {0, SpacingOptions::Undecided}, // No spacing in bit slicing/indexing
+      },
+      {
+          // [a+b]
+          [](){
+            auto style = DefaultStyle;
+            style.compact_non_declaration_ranges = false;
+            return style;
+          }(),
+          verilog_tokentype::SymbolIdentifier,
+          "a",
+          "",  // no spaces originally
+          '+',
+          "+",
+          {NodeEnum::kDimensionScalar},
+          {NodeEnum::kDimensionScalar},
+          {0, SpacingOptions::Undecided},
+      },
+      {
+          // [a +b]
+          [](){
+            auto style = DefaultStyle;
+            style.compact_non_declaration_ranges = false;
+            return style;
+          }(),
+          verilog_tokentype::SymbolIdentifier,
+          "a",
+          " ",  // 1 space originally
           '+',
           "+",
           {NodeEnum::kDimensionScalar},
           {NodeEnum::kDimensionScalar},
-          {1, SpacingOptions::Undecided},  // limit to 1
+          {1, SpacingOptions::Undecided}, // do not compact
+      },
+      {
+          // [a  +b]
+          [](){
+            auto style = DefaultStyle;
+            style.compact_non_declaration_ranges = false;
+            return style;
+          }(),
```

Comment:
To reduce repetition, let's pull this into its own static initializer:

```C++
static const CompactIndexSelectionStyle = [](){...}();
```

Reply:
Done.